src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,85 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+import React, { useEffect, useState } from 'react';
+import { Todo } from '../../types/Todo';
+
+function filterTodos(
+  todos: Todo[],
+  filterMethod: string,
+  query: string,
+): Todo[] {
+  let copy = [...todos];
+
+  if (query) {
+    (copy = [...todos]
+      .filter(todo => todo.title
+        .toLowerCase()
+        .includes(query.toLowerCase())));
+  }
+
+  if (filterMethod === 'active') {
+    (copy = copy.filter(todo => !todo.completed));
+  }
+
+  if (filterMethod === 'completed') {
+    (copy = copy.filter(todo => todo.completed));
```

Comment:
Would be great to use Enum instead of the hardcoded values. Also, switch case might be useful here